cpp/box/export_Box.h

```diff
@@ -13,32 +13,32 @@ namespace freud { namespace box { namespace wrap {
 template<typename T, typename shape = nanobind::shape<-1, 3>>
 using nb_array = nanobind::ndarray<T, shape, nanobind::device::cpu, nanobind::c_contig>;
 
-void makeAbsolute(std::shared_ptr<Box> box, nb_array<float, nanobind::shape<-1, 3>> vecs,
+void makeAbsolute(const std::shared_ptr<Box>& box, nb_array<float, nanobind::shape<-1, 3>> vecs,
```

Comment:
Passing shared pointers by const reference? That doesn't seem like the standard convention.

Reply:
Yes, it seems weird - but `clang-tidy` insists: https://github.com/glotzerlab/freud/actions/runs/9648239399/job/26608946198#step:6:151

It makes sense when you think about it. This function doesn't store the `shared_ptr` anywhere to hold on to the reference, so why not borrow the reference from the caller?
https://stackoverflow.com/questions/74329949/clang-tidy-and-boostintrusive-ptr